lib/utils/qemu.ts

```diff
@@ -148,18 +155,24 @@ export async function installQemuIfNeeded(
 ): Promise<boolean> {
 	// call platformNeedsQemu() regardless of whether emulation is required,
 	// because it logs useful information
-	const needsQemu = await platformNeedsQemu(docker, logger);
+	const needsQemu = await platformNeedsQemu(docker, emulated, logger);
 	if (!emulated || !needsQemu) {
 		return false;
 	}
 	const { promises: fs } = await import('fs');
 	const qemuPath = await getQemuPath(arch);
 	try {
+		const stats = await fs.stat(qemuPath);
+		// Earlier versions of the CLI with broken error handling would leave
+		// behind files with size 0. If such a file is found, delete it.
+		if (stats.size === 0) {
+			await fs.unlink(qemuPath);
```

Comment:
This is part of a bug fix whereby the CLI would leave behind zero-length files if the user specified an invalid architecture on the command line for the build/deploy commands, e.g. `--arch arm` instead of `--arch armv7hf`. Examples:
```
# ls -l ~/.balena/bin/
total 5476
-rw-r--r-- 1 root root       0 Dec 17 23:57 qemu-execve-386-v4.0.0+balena2
-rwxr-xr-x 1 root root 5606984 Dec 17 23:30 qemu-execve-armv7hf-v4.0.0+balena2
-rw-r--r-- 1 root root       0 Dec 17 23:57 qemu-execve-i386-v4.0.0+balena2
-rw-r--r-- 1 root root       0 Dec 17 23:57 qemu-execve-x86-v4.0.0+balena2
```

Reply:
The commit that this comment referred to was moved to PR #2207.